File: java/src/test/java/com/circulocromatico/ChromaticCircleTest.java
package com.circulocromatico;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class ChromaticCircleTest {
    @Test
    public void testMixColorsWithBlueFilter() {
        ChromaticCircle circle = new ChromaticCircle(ChromaticCircle.Color.BLUE);
        
        assertEquals(ChromaticCircle.Color.BLUE, circle.mixColors(ChromaticCircle.Color.BLUE));
        assertEquals(ChromaticCircle.Color.GREEN, circle.mixColors(ChromaticCircle.Color.YELLOW));
        assertEquals(ChromaticCircle.Color.VIOLET, circle.mixColors(ChromaticCircle.Color.RED));
        
        assertThrows(IllegalArgumentException.class, () -> 
            circle.mixColors(ChromaticCircle.Color.GREEN)
        );
    }

    @Test
    public void testMixColorsWithRedFilter() {
        ChromaticCircle circle = new ChromaticCircle(ChromaticCircle.Color.RED);
        
        assertEquals(ChromaticCircle.Color.RED, circle.mixColors(ChromaticCircle.Color.RED));
        assertEquals(ChromaticCircle.Color.ORANGE, circle.mixColors(ChromaticCircle.Color.YELLOW));
        assertEquals(ChromaticCircle.Color.VIOLET, circle.mixColors(ChromaticCircle.Color.BLUE));
        
        assertThrows(IllegalArgumentException.class, () -> 
            circle.mixColors(ChromaticCircle.Color.GREEN)
        );
    }

    @Test
    public void testMixColorsWithYellowFilter() {
        ChromaticCircle circle = new ChromaticCircle(ChromaticCircle.Color.YELLOW);
        
        assertEquals(ChromaticCircle.Color.YELLOW, circle.mixColors(ChromaticCircle.Color.YELLOW));
        assertEquals(ChromaticCircle.Color.ORANGE, circle.mixColors(ChromaticCircle.Color.RED));
        assertEquals(ChromaticCircle.Color.GREEN, circle.mixColors(ChromaticCircle.Color.BLUE));
        
        assertThrows(IllegalArgumentException.class, () -> 
            circle.mixColors(ChromaticCircle.Color.VIOLET)
        );
    }

    @Test
    public void testMixColorsWithSameColor() {
        ChromaticCircle circle = new ChromaticCircle(ChromaticCircle.Color.YELLOW);
        
        assertEquals(ChromaticCircle.Color.YELLOW, circle.mixColors(ChromaticCircle.Color.YELLOW));
    }

    @Test
    public void testMixColorsNonExistingMixer() {
        ChromaticCircle circle = new ChromaticCircle(ChromaticCircle.Color.VIOLET);

        IllegalArgumentException illegalArgumentException = assertThrows(IllegalArgumentException.class, () ->
                circle.mixColors(ChromaticCircle.Color.VIOLET)
        );
        assertEquals("No mixer found for filter: VIOLET", illegalArgumentException.getMessage());
    }
}

File: dart/test/chromatic_circle_test.dart
import 'package:test/test.dart';
import '../lib/chromatic_circle.dart';

void main() {
  group('ChromaticCircle', () {
    test('mixColors returns correct colors with blue filter', () {
      final circle = ChromaticCircle(Color.blue);
      
      expect(circle.mixColors(Color.blue), Color.blue);
      expect(circle.mixColors(Color.yellow), Color.green);
      expect(circle.mixColors(Color.red), Color.violet);
      
      expect(() => circle.mixColors(Color.green),
          throwsA(isA<ArgumentError>()));
    });

    test('mixColors returns correct colors with red filter', () {
      final circle = ChromaticCircle(Color.red);
      
      expect(circle.mixColors(Color.red), Color.red);
      expect(circle.mixColors(Color.yellow), Color.orange);
      expect(circle.mixColors(Color.blue), Color.violet);
      
      expect(() => circle.mixColors(Color.green),
          throwsA(isA<ArgumentError>()));
    });

    test('mixColors returns correct colors with yellow filter', () {
      final circle = ChromaticCircle(Color.yellow);
      
      expect(circle.mixColors(Color.yellow), Color.yellow);
      expect(circle.mixColors(Color.red), Color.orange);
      expect(circle.mixColors(Color.blue), Color.green);
      
      expect(() => circle.mixColors(Color.violet),
          throwsA(isA<ArgumentError>()));
    });

    test('mixColors returns same color when filter matches input', () {
      final circle = ChromaticCircle(Color.green);
      
      expect(circle.mixColors(Color.green), Color.green);
    });
  });
}
